src/ai/backend/common/bgtask.py

```diff
@@ -188,7 +201,9 @@ async def push_bgtask_events(
                 finally:
                     return resp
         finally:
-            self.task_update_queues.remove(my_queue)
+            self.task_update_queues[task_id].remove(my_queue)
+            if len(self.task_update_queues[task_id]) == 0:
+                del self.task_update_queues[task_id]
```

Comment:
Is there a consideration for concurrency (e.g. handling multiple `push_bgtask_events()` calls to same task_id)?

Reply:
I missed it. I added the critical section on the creation and deletion of the set with asyncio.Lock.
Please check the latest commit. (The rebase messed up the commit history;)